addons/purchase_product_matrix/views/purchase_views.xml

```diff
@@ -7,7 +7,8 @@
         <field name="inherit_id" ref="purchase.purchase_order_form"/>
         <field name="arch" type="xml">
             <xpath expr="//tree/field[@name='product_id']" position="attributes">
-                <attribute name="invisible">1</attribute>
+                <attribute name="string">Product Variant</attribute>
```

Comment:
this should probably be added to addons/purchase_product_matrix/i18n/purchase_product_matrix.pot